File: dotnet-nunit-selenium/TestBase.cs
public class TestBase
{
    protected bool _testPassed;
    protected string _reportFilePath = Path.Combine(Directory.GetCurrentDirectory(),"TestResultsSummary.txt");
    
    const string _site1 = "https://juanfonsecagl.github.io/SelfHealingLocatorsDemo/demo-site/form-button-label.html";
    const string _site2 = "https://juanfonsecagl.github.io/SelfHealingLocatorsDemo/demo-site/form-button-label2.html";
    const string _site3 = "https://juanfonsecagl.github.io/SelfHealingLocatorsDemo/demo-site/form-button-label3.html";

    public static object[] TestData =
    {
        new object[] { "TC1", _site1,    "//input[@id='messageNew']",                "//button[@id='changer']",  "correct" },    // TC1
        new object[] { "TC2", _site2,    "//input[@id='messageNew']",                "//a[@id='pusher']",        "correct" },    // TC2
        new object[] { "TC3", _site1,    "//input[@id='messageNew']",                "//a[@id='pusher']",        "broken" },     // TC3
        new object[] { "TC4", _site2,    "//input[@id='messageNew']",                "//button[@id='changer']",  "broken" },     // TC4

        // TestSuite #2: different placeholder for input (scenario proposed by Healenium)
        new object[] { "TC5", _site1,   "//input[@placeholder='Message']",          "//button[@id='changer']",   "correct" },    // TC5
        new object[] { "TC6", _site3,   "//input[@placeholder='Enter some text']",  "//button[@id='changer']",   "correct" },    // TC6
        new object[] { "TC7", _site1,   "//input[@placeholder='Enter some text']",  "//button[@id='changer']",   "broken" },     // TC7
        new object[] { "TC8", _site3,   "//input[@placeholder='Message']",          "//button[@id='changer']",   "broken" }      // TC8
    };
}

File: java-testng-selenium/src/test/java/selfhealinglocators/Tests/TestBase.java
package selfhealinglocators.Tests;

import org.testng.annotations.DataProvider;
import org.openqa.selenium.By;

public class TestBase 
{
    String baseSite = "https://juanfonsecasolis-gorillalogic.github.io/SelfHealingLocatorsDemo/demo-site";
    String site1 = baseSite+"/demo-site1/form-button-label.html";
    String site2 = baseSite+"/demo-site1/form-button-label2.html";
    String site3 = baseSite+"/demo-site1/form-button-label3.html";
    String site4 = baseSite+"/demo-site2/reveal-link1.html";
    String site5 = baseSite+"/demo-site2/reveal-link2.html";

    @DataProvider(name = "testDataSite123")
    public Object[][] testDataSite123() {
        return new Object[][] 
        {
            // TestSuite #1: different button implementation (scenario proposed by Test Rigor)
            { "TC1", site1,    By.xpath("//input[@id='messageNew']"),                By.xpath("//button[@id='changer']"),  "valid" },
            { "TC2", site2,    By.xpath("//input[@id='messageNew']"),                By.xpath("//a[@id='pusher']"),        "valid" },
            { "TC3", site1,    By.xpath("//input[@id='messageNew']"),                By.xpath("//a[@id='pusher']"),        "broken" },
            { "TC4", site2,    By.xpath("//input[@id='messageNew']"),                By.xpath("//button[@id='changer']"),  "broken" },

            // TestSuite #2: different placeholder for input (scenario proposed by Healenium)
            { "TC5", site1,    By.xpath("//input[@placeholder='Message']"),          By.xpath("//button[@id='changer']"),   "valid" },
            { "TC6", site3,    By.xpath("//input[@placeholder='Enter some text']"),  By.xpath("//button[@id='changer']"),   "valid" },
            { "TC7", site1,    By.xpath("//input[@placeholder='Enter some text']"),  By.xpath("//button[@id='changer']"),   "broken" },
            { "TC8", site3,    By.xpath("//input[@placeholder='Message']"),          By.xpath("//button[@id='changer']"),   "broken" }
        };
    }

    @DataProvider(name = "testDataSite45")
    public Object[][] testDataSite45() {
        return new Object[][] 
        {
            // TestSuite #1: different button implementation (scenario proposed by Test Rigor)
            { "TC9", site4,    By.xpath("//div[@class='container']/div/div/button"),  "valid" },
            { "TC10", site4,    By.xpath("//div[@class='container']/button"),        "valid" },
            { "TC11", site5,    By.xpath("//div[@class='container']/button"),        "broken" },
            { "TC12", site5,    By.xpath("//div[@class='container']/div/div/button"),  "broken" },
        };
    }
}
